docs/intro/from-ground-up.rst

```diff
@@ -219,542 +183,70 @@ objects, so ``web-poet`` provides a shortcut for it: inherit from
                 "url": self.response.url,
                 "title": self.response.css("h1").get(),
-                "description": self.response.css("#product_description+ p").get().strip(),
-                # ...
-            }
-
-There are pros and cons for using classes vs functions for writing
-such extraction code, but the distinction is not that important;
-web-poet uses classes at the moment.
-
-to_item() method
-================
-
-It is common to have Page Objects for a web page where a single main
-data record needs to be extracted (e.g. book information in our example).
-``web-poet`` standardizes this, by asking to name a method implementing the
-extraction ``to_item``.
-
-As the method name is now standardized, the code which creates a Page Object
-instance can now work for other Page Objects like that. For example, you can
-have ``ToscrapeBookPage`` and ``BamazonBookPage`` classes, and
-
-.. code-block:: python
-
-    def get_item(page_cls: WebPage, response: HttpResponse) -> dict:
-        page = page_cls(response=response)
-        return page.to_item()
-
-would work for both.
-
-But wait. Before the example was converted to ``web-poet``, we were getting
-it for free:
-
-.. code-block:: python
-
-    def get_item(extract_func, response: HttpResponse) -> dict:
-        return extract_func(url=response.url, text=response.text)
-
-No need to agree on ``to_item`` name or have a base class.
-Why bother with classes then?
-
-Classes for web scraping code
-=============================
-
-A matter of preference. Functions are great, too.
-Classes sometimes can make it a easier to organize web scraping code.
-For example, we can extract logic for different attributes into properties:
-
-.. code-block:: python
-
-    class BookPage(WebPage):
-        """
-        A book page on http://books.toscrape.com website, e.g.
-        http://books.toscrape.com/catalogue/a-light-in-the-attic_1000/index.html
-        """
-
-        @property
-        def title(self):
-            return self.response.css("h1").get()
-
-        @property
-        def description(self):
-            return self.response.css("#product_description+ p").get().strip()
-
-        def to_item(self):
-            return {
-                "url": self.response.url,
-                "title": self.title,
-                "description": self.description,
-                # ...
             }
 
-It might be easier to read the code written this way. Also, this style
-allows to extract only some of the attributes - if you don't need
-the complete to_item() output, you still can access individual properties.
-
-web-poet provides a small framework to simplify writing Page Objects
-in this style; see :ref:`fields`. The example above can be simplified
-using web-poet fields - there is no need to write ``to_item`` boilerplate:
+At this point you may be wondering why web-poet requires you to write a class
+with a ``to_item`` method rather than a function. The answer is flexibility.
```

Comment:
There is a second reason, related to andi: classes can be used as types in Python, while functions can't. So, you can type annotate "I need an instance of this class", but you can't say "I want this function specifically". Currently typing is used for dependency injection.

If not for dependency injection, https://github.com/scrapinghub/web-poet/pull/44/files wouldn't need to create a class from a function.

I'm not sure if we should go into explaining all this in this document though. Could be worth adding a note, but I'm not sure :)

Reply:
https://github.com/scrapinghub/web-poet/pull/152/commits/fcd364061034500a74d32dfd315572a05837e3f8